scripts/kibana/validate-ts-interfaces-against-apm-server-sample-docs/scripts/download-sample-docs.ts

```diff
@@ -9,55 +9,84 @@ const [owner = 'elastic', branch = 'master'] = process.argv.slice(2);
-    interfacePath: '../apm-ui-interfaces/raw/transaction_raw',
-    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationTransactions.approved.json`
+    name: 'transactions',
+    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationTransactions.approved.json`,
+    getFileContent: data => {
+      return `import { TransactionRaw } from '../apm-ui-interfaces/raw/transaction_raw';
+      import { AllowUnknownProperties } from '../../scripts/helpers';
+      export const sampleDoc:AllowUnknownProperties<TransactionRaw>[] = ${JSON.stringify(
+        data.events
+      )}`;
+    }
   },
   {
-    interfaceName: 'ErrorRaw',
-    interfacePath: '../apm-ui-interfaces/raw/error_raw',
-    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationErrors.approved.json`
+    name: 'errors',
+    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationErrors.approved.json`,
+    getFileContent: data => {
+      return `import { ErrorRaw } from '../apm-ui-interfaces/raw/error_raw';
+      import { AllowUnknownProperties } from '../../scripts/helpers';
+      export const sampleDoc:AllowUnknownProperties<ErrorRaw>[] = ${JSON.stringify(
+        data.events
+      )}`;
+    }
   },
   {
-    interfaceName: 'MetricRaw',
-    interfacePath: '../apm-ui-interfaces/raw/metric_raw',
-    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationMetricsets.approved.json`
+    name: 'metrics',
+    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationMetricsets.approved.json`,
+    getFileContent: data => {
+      return `import { MetricRaw } from '../apm-ui-interfaces/raw/metric_raw';
+      import { AllowUnknownProperties } from '../../scripts/helpers';
+      export const sampleDoc:AllowUnknownProperties<MetricRaw>[] = ${JSON.stringify(
+        data.events
+      )}`;
+    }
+  },
+  {
+    name: 'minimal',
+    url: `https://raw.githubusercontent.com/elastic/apm-server/${branch}/beater/test_approved_es_documents/TestPublishIntegrationMinimalEvents.approved.json`,
+    getFileContent: data => {
+      return `import { SpanRaw } from '../apm-ui-interfaces/raw/span_raw';
+      import { TransactionRaw } from '../apm-ui-interfaces/raw/transaction_raw';
+      import { ErrorRaw } from '../apm-ui-interfaces/raw/error_raw';
+      import { MetricRaw } from '../apm-ui-interfaces/raw/metric_raw';
+      import { AllowUnknownProperties } from '../../scripts/helpers';
+      export const sampleDoc: AllowUnknownProperties<
+        SpanRaw | TransactionRaw | ErrorRaw | MetricRaw
+      >[] = ${JSON.stringify(data.events)}`;
+    }
   }
```

Comment:
This is the meat of the PR: this adds a new section that will download a [minimal doc](https://github.com/elastic/apm-server/blob/master/beater/test_approved_es_documents/TestPublishIntegrationMinimalEvents.approved.json) from APM Server and test it against the typescript interfaces in Kibana